### DISCUSSION TITLE
Spine Animation Issue

### DISCUSSION THREAD

**Discussion by arungowda09** [axmolengine/axmol#2264] [Q&A] [answered]
While updating from cocos2dx 3.17.2 to axmol spine animations version issues

is there any way to use same files Or we have re export to 4.1 version ?

Assert failed: Skeleton version 3.6.53 does not match runtime version 4.1

**Comment by rh101** [comment, marked as answer]
Go here: https://github.com/axmolengine/spine-axmol

Select the correct branch for your version of spine.

Delete the `{axmol}/extensions/spine` folder, then copy the new one from the above branch in its place.

Note that if the branch you need to use doesn't compile, it means that it needs to be updated to work with Axmol 2.1.2.  You can see what possible changes are needed in this example:

https://github.com/axmolengine/spine-axmol/commit/83f114f71a28a46d16ff389aee81dc44ce51b8c7

**Reply by arungowda09** [reply]
Thank you, @rh101, for your response. The **spine36** branch indeed resolves the issue with version axmol 2.2.1. However, I encountered some additional compile issues beyond those mentioned in [this commit](https://github.com/axmolengine/spine-axmol/commit/83f114f71a28a46d16ff389aee81dc44ce51b8c7).

I have resolved these issues on my end.

**Reply by rh101** [reply]
> However, I encountered some additional compile issues beyond those mentioned in [this commit](https://github.com/axmolengine/spine-axmol/commit/83f114f71a28a46d16ff389aee81dc44ce51b8c7).

Are you able to create a pull request with the changes for branch spine36 on the spine-axmol repo?

**Reply by arungowda09** [reply]
To create a pull request directly on the spine-axmol repo, I would need to be a collaborator with write access. If I'm not a collaborator, I can fork the repository, make the changes, and submit the pull request from my forked repo. Let me know which option works for you!

**Reply by rh101** [reply]
> To create a pull request directly on the spine-axmol repo, I would need to be a collaborator with write access. If I'm not a collaborator, I can fork the repository, make the changes, and submit the pull request from my forked repo. Let me know which option works for you!

The correct process is as follows:

1 - Fork the repo
2 - Create a branch based on `spine36` (avoid working on branch `spine36` directly, even in your own repo)
3 - Make the changes and commit them against your new branch
4 - Create a pull request from your repo and the new branch you created against the primary repo and branch `spine36` (not against your repo)

If you have issues, then just make the changes to your own repo/branch and link it in here, since we can review the changes directly on your repo.
